dali/pipeline/operators/reader/reader_op.h

```diff
@@ -186,7 +198,8 @@ class DataReader : public Operator<Backend> {
       // if we've consumed all samples in this batch, reset state and stop
       if (samples_processed_.load() == Operator<Backend>::batch_size_) {
         prefetch_ready_workers_ = false;
-        prefetched_batch_ready_ = false;
+        curr_batch_consumer_ = NextIdx(curr_batch_consumer_, consumer_cycle_);
```

Comment:
It should be protected by prefetch_access_mutex_

Reply:
should be fixed now